docs/Style.thy

```diff
@@ -0,0 +1,182 @@
+(*
+ * Copyright 2021, Data61, CSIRO
+ *
+ * This software may be distributed and modified according to the terms of
+ * the BSD 2-Clause license. Note that NO WARRANTY is provided.
+ * See "LICENSE_BSD2.txt" for details.
+ *
+ * @TAG(DATA61_BSD)
+ *)
+
+chapter \<open>An Isabelle Syntax Style Guide\<close>
+
+theory Style
+  imports Main
+begin
+
+(* This style guide is heavily influenced by Gerwin's Style Guide for Isabelle/HOL. See parts 1
+   and 2 here:
+
+   * https://proofcraft.org/blog/isabelle-style.html
+   * https://proofcraft.org/blog/isabelle-style-part2.html
+
+   This guide may repeat some parts of Gerwin's guide and should not conflict in any way.
+ *)
+
+section \<open>General Principles\<close>
+
+(* TO DO: What advice should go here? *)
+
+
+section \<open>Definitions\<close>
+
+(* Definitions should be written over 2 lines in the following style.
+
+   Note that indentation is 2 spaces and the definitions do not run past 100 character line
+   limit. Note also that the \equiv symbol is used.
+ *)
+
+definition hd_opt :: "'a list \<Rightarrow> 'a option" where
+  "hd_opt \<equiv> case_list None (\<lambda>h t. Some h)"
```

Comment:
It'd be nicer to use an example that has a parameter on the lhs, just to give a more familiar pattern.